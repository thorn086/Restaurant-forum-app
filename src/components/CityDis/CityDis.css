.delete-btn{
    margin: 1rem;
    background-color: rgb(98, 98, 250);
    width: 7rem;
    height: 2rem;
    border-radius: 10%;
    color: white;
    font-size: 12px;
    font-family: 'Fjalla One', sans-serif;
    letter-spacing: 1px;
}
.state-name{
    margin: 1rem 0;
    color: white;
    text-transform: uppercase;
    text-shadow: black 1px 1px;
}
.error-name,.error_title, .author_userId{
    color: white;
    text-transform: uppercase;
    text-shadow: black 1px 1px;
}
.error_title{
    font-size: 25px;
    
}
@media (min-width: 760px){
   
    .delete-btn{
        font-size: 14px;
    }
    .state-items.error{
        width: 20rem;
        height: 8rem;
        margin:3rem 12rem ;
        padding-top: 3rem;
    }
   
}
@media (min-width: 1020px){
    .state_title{
        font-size: 50px;
    }
    .delete-btn{
        font-size: 16px;
    }
    .back-btn.city{
        margin: 2rem 0 2rem 12rem;
    }
    .state-items.error{
        
        margin:2rem 21rem ;
        padding-top: 3rem;
    }
}
@media (min-width: 1350px){
    .city-nav{
        width: 80%;
        margin: auto;
    }
    .state-items.error{
        margin:2rem 22rem ;
        padding-top: 3rem;
    }
    .back-btn.city{
        margin: 2rem 0 2rem 13rem;
    }
}
@media (min-width: 2560px){
    .state_title{
        font-size: 65px;
    }
    .state-items.error{
        width: 25rem;
        height: 10rem;
        margin:2rem 25rem ;
        padding-top: 3rem;
    }
    .back-btn.city{
        margin: 4rem 0 4rem 17rem;
    }
}