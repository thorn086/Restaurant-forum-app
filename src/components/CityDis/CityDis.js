import React from 'react'
import { withRouter, NavLink } from 'react-router-dom'
import AddCity from '../../components/AddCity/AddCity'
import restForumContext from '../../context'
import config from '../../config'
import TokenService from '../../services/token-services'
import './CityDis.css'


class CityDis extends React.Component {
    static contextType = restForumContext
    state={
        show:false
        
    }

    handleDeleteCity=event=>{
        event.preventDefault()
        const citySelected = event.target.value
        
        fetch(`${config.API_ENDPOINT}/states/${citySelected}`,{
            method:'DELETE',
            headers:{
            'content-type':'application/json',
            'authorization':`bearer ${TokenService.getAuthToken()}`
        },
    })
        .then(result => {
            if (!result.ok)
                return result.json().then(event => Promise.reject(event))
        })
        .then(()=> {
           this.context.deleteCity(citySelected)
            
        })
        .catch(error => {
            console.log({ error })
        })
    }

    hasError() {
        const { error } = this.props
        return (
            <ul className='state-list-items error'>
                <li className='state-items error'>
                <p className='error_title error'>Sorry</p>
                </li>
                <li key={error} className='state-items error'>
                    <p className='error-name'>{error}</p>
                </li>
            </ul>
        )
    }

   
    hasNoError() {
        const { cities } = this.context
        return (
            <ul className='state-list-items'>
                {cities.map((newCity, i) =>{
                    return (
                        <li key={i} className='state-items'>
                            <NavLink className='state-list-link'
                                to={`/city/${newCity.id}`}>
                                <h3 className='state-name'>
                                    {newCity.name}
                                </h3>
                            </NavLink>
                            <p className='author_userId'> Added By User: {newCity.author}</p> 
                            <button value={newCity.id} className='delete-btn' type='button' onClick={this.handleDeleteCity}>Delete</button>

                        </li>)
                })}
            </ul>
        )
    }
    showCityAdd = () => {
        this.setState({
            show:!this.state.show
        })
    }


    handleCityTitle() {
        const {show} =this.state
        if(show === false){
            return (
                'Add City'
            )
        }else{
            return(
                'Cancel'
            )
        }

    }
    render() {
        const { error } = this.props
       
        return (
            <div className='state_list'>
                {(error === null) ? this.hasNoError() : this.hasError()}
                {this.state.show ? <AddCity show={this.showCityAdd}/> :null}
                <button className=' back-btn city' tag='button' onClick={()=> this.props.history.goBack()}>Back</button>
                <button className=' back-btn city' tag='button' onClick={()=> this.showCityAdd()}>{this.handleCityTitle()}</button>
                
            </div>
        )
    }
}
export default withRouter(CityDis)