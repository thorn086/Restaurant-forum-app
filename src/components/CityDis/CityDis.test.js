import React from 'react';
import CityDis from './CityDis';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('CityDis', () => {
    const cityDis = shallow(<CityDis/>);
  
    it('renders properly', () => {
        expect(cityDis).toMatchSnapshot();
    });
});