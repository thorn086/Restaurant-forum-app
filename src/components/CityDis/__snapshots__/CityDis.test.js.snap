// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityDis renders properly 1`] = `ShallowWrapper {}`;
